src/Node/ForNode.php

```diff
@@ -25,7 +25,7 @@ class ForNode extends Node
 {
     private $loop;
 
-    public function __construct(AssignNameExpression $keyTarget, AssignNameExpression $valueTarget, AbstractExpression $seq, $ifexpr = null, Node $body, Node $else = null, int $lineno, string $tag = null)
+    public function __construct(AssignNameExpression $keyTarget, AssignNameExpression $valueTarget, AbstractExpression $seq, $ifexpr, Node $body, ?Node $else, int $lineno, ?string $tag = null)
```

Comment:
What about adding default values instead? That way, we could merge it in 1.x.

Reply:
It was refused in #3327 but I will do that tomorrow